frontend/packages/pipelines-plugin/src/components/import/pipeline/pipeline-template-utils.ts

```diff
@@ -67,6 +72,11 @@ export const createPipelineForImportFlow = async (
 export const createPipelineRunForImportFlow = async (pipeline: Pipeline): Promise<PipelineRun> => {
   const pipelineInitialValues: StartPipelineFormValues = {
     ...convertPipelineToModalData(pipeline),
+    workspaces: (pipeline.spec.workspaces || []).map((workspace: PipelineWorkspace) => ({
```

Comment:
is it better to have a null check? pipeline.spec?.workspaces

Reply:
I have added a fallback  `|| []`  above, so it will not break.

Reply:
sure but if a pipeline doesn't have specs yet, could it through cannot read property `workspaces` of undefined

Reply:
You can't create a Pipeline without a `spec`:
![image](https://user-images.githubusercontent.com/8126518/99844663-71614200-2b41-11eb-9a25-22a3747987c5.png)